gems/smithy/lib/smithy/anvil/client/templates/shapes.erb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+# This is generated code!
+
+module <%= namespace %>
+  module Shapes
+    include Smithy::Client::Shapes
+
+<%# Shapes definition -%>
+  <%- @shapes.each do |shape| -%>
+    <%- params = ["id: \"#{shape.id}\""] -%>
+    <%- params << "type: Types::#{shape.name}" if shape.typed -%>
+    <%- params << "traits: #{shape.traits}" unless shape.traits.empty? -%>
+    <%= "#{shape.name} = #{shape.shape_type}.new(" %><%= params.join(', ') %>)
+  <%- end -%>
+<%# Shape members definition %>
+  <%- shapes_with_members.each do |shape| -%>
+    <%- shape.members.each do |member| -%>
+    <%- params = [] -%>
+    <%- params << ["\"#{member.name}\""] unless %w[ListShape MapShape].include? shape.shape_type -%>
+    <%- params << [member.shape] -%>
+    <%- params << "traits: #{member.traits}" unless member.traits.empty? -%>
+    <%- case shape.shape_type -%>
+    <%- when 'ListShape' -%>
+    <%= "#{shape.name}.set_member(" %><%= params.join(', ') %>)
+    <%- when 'MapShape' -%>
+    <%= "#{shape.name}.set_member_#{member.name}(" %><%= params.join(', ') %>)
+    <%- else -%>
+    <%= "#{shape.name}.add_member(" %><%= params.join(', ') %>)
+    <%- end -%>
+    <%- end -%>
+  <%- end -%>
+<%# Schema definition %>
+    SCHEMA = Smithy::Client::Schema.new do |schema|
+      schema.service = ServiceShape.new(
+        id: "<%= @service_shape.id %>",
+        version: "<%= @service_shape.version %>",
+        traits: <%= @service_shape.traits %>
+      )
+    <%- @operation_shapes.each do |shape| %>
+      schema.add_operation(:<%= shape.name %>, OperationShape.new do |operation|
+        operation.id = "<%= shape.id %>"
+        operation.input = <%= shape.input %>
+        operation.output = <%= shape.output %>
+       <%- unless shape.traits.empty? -%>
+        operation.traits = <%= shape.traits -%>
+       <% end -%>
+       <%- unless shape.errors.empty? -%>
+       <% shape.errors.each do |err| %>
+        operation.errors << <%= err -%>
```

Comment:
Should not end in -%> for an <%=. Please inspect rendered code.

Reply:
I was under the impression that `<%= EXPRESSION %>` could end with `-%> ` to trim the following line break.

Rendered code based on `<%= err -%>`:
```
schema.add_operation(:get_city, OperationShape.new do |operation|
  operation.id = "example.weather#GetCity"
  operation.input = GetCityInput
  operation.output = GetCityOutput
  operation.traits = {"smithy.api#readonly"=>{}}              
  operation.errors << NoSuchResource             
end)
```
Where as,  `<%= err %>` will produce:
```
schema.add_operation(:get_city, OperationShape.new do |operation|
  operation.id = "example.weather#GetCity"
  operation.input = GetCityInput
  operation.output = GetCityOutput
  operation.traits = {"smithy.api#readonly"=>{}}              
  operation.errors << NoSuchResource
             
end)
```

Reply:
I believe that is because your outer loop does not have dash trimming.

Reply:
Yeah that fixes the above operation and the following operations will have a weird formatting so I had to keep that dash.